Gem/Code/Source/Components/Multiplayer/EnergyBallComponent.cpp

```diff
@@ -5,120 +5,142 @@
     }
-#endif
 
-    void EnergyBallComponentController::HideEnergyBall()
+    void EnergyBallComponentController::CheckForCollisions()
     {
-        Physics::RigidBodyRequestBus::Event(GetEntityId(), &Physics::RigidBodyRequestBus::Events::DisablePhysics);
-
-        // move self and increment resetCount to prevent transform interpolation
-        AZ::TransformBus::Event(GetEntityId(), &AZ::TransformBus::Events::SetWorldTranslation, AZ::Vector3::CreateAxisZ(-1000.f));
-        GetNetworkTransformComponentController()->ModifyResetCount()++;
-    }
+        const AZ::Vector3& position = GetEntity()->GetTransform()->GetWorldTM().GetTranslation();
+        const HitEffect& effect = GetHitEffect();
 
-    void EnergyBallComponentController::OnCollisionBegin(const AzPhysics::CollisionEvent& collisionEvent)
-    {
-        if (collisionEvent.m_body2)
+        IntersectResults results;
+        const ActivateEvent activateEvent{ GetEntity()->GetTransform()->GetWorldTM(), position, Multiplayer::InvalidNetEntityId, Multiplayer::InvalidNetEntityId };
+        GatherEntities(GetGatherParams(), activateEvent, m_filteredNetEntityIds, results);
+        if (!results.empty())
         {
-            AZ::Entity* target = nullptr;
-            AZ::ComponentApplicationBus::BroadcastResult(target,
-                &AZ::ComponentApplicationBus::Events::FindEntity, collisionEvent.m_body2->GetEntityId());
-            if (target)
+            bool shouldTerminate = false;
+            for (const IntersectResult& result : results)
             {
-#if AZ_TRAIT_SERVER
-                if (NetworkHealthComponent* health = target->FindComponent<NetworkHealthComponent>())
+                Multiplayer::ConstNetworkEntityHandle handle = Multiplayer::GetMultiplayer()->GetNetworkEntityManager()->GetEntity(result.m_netEntityId);
+                if (handle.Exists())
                 {
-                    health->SendHealthDelta(aznumeric_cast<float>(-m_armorDamage));
+                    shouldTerminate = true;
+
+                    // Presently set to 1 until we capture falloff range
+                    float hitDistance = 1.f;
+                    float maxDistance = 1.f;
+                    float damage = effect.m_hitMagnitude * powf((effect.m_hitFalloff * (1.0f - hitDistance / maxDistance)), effect.m_hitExponent);
+
+                    // Look for physics rigid body component and make impact updates
+                    if (Multiplayer::NetworkRigidBodyComponent* rigidBodyComponent = handle.GetEntity()->FindComponent<Multiplayer::NetworkRigidBodyComponent>())
+                    {
+                        const AZ::Vector3 explosionCentre = position;
+                        const AZ::Vector3 hitObject = handle.GetEntity()->GetTransform()->GetWorldTM().GetTranslation();
+                        const AZ::Vector3 impulse = (hitObject - position).GetNormalized() * damage * sv_EnergyBallImpulseScalar;
```

Comment:
when a player runs at an energy ball headed straight for them, if the intersection tunnels through the player, the impulse might accelerate them forward

Reply:
Right now this doesn't affect players, only dynamic rigids in the world (like the boxes in the video). Players only receive damage. We could try adding velocity to the players as well if we wanted a quick and dirty simulation of knock-back, but I decided not to press my luck with this PR..